### PR TITLE
[TTAHUB-2885] Monitoring Findings

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change

- Update ETL import maintenance job for monitoring data to include pulling in data for findings.
- Add new tables and models to support the new data
- Update seed data to account for some of the new structure 

## How to test

1. Run ./bin/load-test-db
2. Enter the docker container running the backend
3. Run yarn import:system download 1
4. Run yarn import:system process 1
5. Verify no errors were thrown
6. Verify data is present in the table "MonitoringStandards"

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2885

### Every PR

- [x] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [x] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — APPROVED** [review]

**Comment by kryswisnaskas** [comment]
Can tests for the new data also be added in this PR? 🌮

**Comment by GarrettEHill** [comment]
> Can tests for the new data also be added in this PR? 🌮
The new data captured is already covered by the current tests.

**Review by AdamAdHocTeam — COMMENTED** [review]
Might be a dumb question, but it seems like we are adding alot (10) of models/tables for this new monitoring data. Is there any way we can reduce the number of newly created items?

Maybe some sort of generic data warehouse?

**Comment by GarrettEHill** [comment]
> Maybe some sort of generic data warehouse?

No, we need to keep in sync with the data coming in to be able to maintain it correctly.

File: src/models/tests/monitoring/monitoring.test.js [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+    // Check associations
+    const statusLink = await MonitoringReviewStatusLink.findOne({ where: { statusId: 2 } });
+    expect(statusLink).not.toBe(null);
+
+    await MonitoringReviewStatus.destroy({
+      where: { statusId: 2 },
+      fource: true,
+    });
+    await MonitoringReviewStatusLink.destroy({ where: { statusId: 2 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:493]:
```suggestion
      force: true,
    });
    await MonitoringReviewStatusLink.destroy({ where: { statusId: 2 }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+
+    // Check associations
+    const standardLink = await MonitoringStandardLink.findOne({ where: { standardId: 102 } });
+    expect(standardLink).not.toBe(null);
+
+    await MonitoringStandard.destroy({
+      where: { standardId: 102 },
+      fource: true,
+    });
+    await MonitoringStandardLink.destroy({ where: { standardId: 102 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:540]:
```suggestion
      force: true,
    });
    await MonitoringStandardLink.destroy({ where: { standardId: 102 }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+
+    const grantLink = await GrantNumberLink.findOne({ where: { grantNumber: 'Grant123' } });
+    expect(grantLink).not.toBe(null);
+
+    await MonitoringClassSummary.destroy({
+      where: { grantNumber: 'Grant123', reviewId: 'Review123' },
+      fource: true,
+    });
+    await GrantNumberLink.destroy({ where: { grantNumber: 'Grant123' }, fource: true });
+    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review123' }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:69]:
```suggestion
      force: true,
    });
    await GrantNumberLink.destroy({ where: { grantNumber: 'Grant123' }, force: true });
    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review123' }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+
+    const findingLink = await MonitoringFindingLink.findOne({ where: { findingId: '123' } });
+    expect(findingLink).not.toBe(null);
+
+    await MonitoringFinding.destroy({
+      where: { statusId: 1, findingId: '123' },
+      fource: true,
+    });
+    await MonitoringFindingStatusLink.destroy({ where: { statusId: 1 }, fource: true });
+    await MonitoringFindingLink.destroy({ where: { findingId: '123' }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:116]:
```suggestion
      force: true,
    });
    await MonitoringFindingStatusLink.destroy({ where: { statusId: 1 }, force: true });
    await MonitoringFindingLink.destroy({ where: { findingId: '123' }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+    const granteeLink = await MonitoringGranteeLink.findOne({ where: { granteeId: 'G123' } });
+    expect(granteeLink).not.toBe(null);
+
+    await MonitoringFindingGrant.destroy({
+      where: { statusId: 2, findingId: 'F123', granteeId: 'G123' },
+      fource: true,
+    });
+    await MonitoringFindingLink.destroy({ where: { findingId: 'F123' }, fource: true });
+    await MonitoringFindingStatusLink.destroy({ where: { statusId: 2 }, fource: true });
+    await MonitoringGranteeLink.destroy({ where: { granteeId: 'G123' }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:172]:
```suggestion
    await MonitoringFindingGrant.destroy({
      where: { statusId: 2, findingId: 'F123', granteeId: 'G123' },
      force: true,
    });
    await MonitoringFindingLink.destroy({ where: { findingId: 'F123' }, force: true });
    await MonitoringFindingStatusLink.destroy({ where: { statusId: 2 }, force: true });
    await MonitoringGranteeLink.destroy({ where: { granteeId: 'G123' }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+    const statusLink = await MonitoringFindingHistoryStatusLink.findOne({ where: { statusId: 1 } });
+    expect(statusLink).not.toBe(null);
+
+    await MonitoringFindingHistory.destroy({
+      where: { statusId: 1, findingId: 'F124', reviewId: 'R123' },
+      fource: true,
+    });
+    await MonitoringReviewLink.destroy({ where: { reviewId: 'R123' }, fource: true });
+    await MonitoringFindingLink.destroy({ where: { findingId: 'F124' }, fource: true });
+    await MonitoringFindingHistoryStatusLink.destroy({ where: { statusId: 1 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:229]:
```suggestion
      force: true,
    });
    await MonitoringReviewLink.destroy({ where: { reviewId: 'R123' }, force: true });
    await MonitoringFindingLink.destroy({ where: { findingId: 'F124' }, force: true });
    await MonitoringFindingHistoryStatusLink.destroy({ where: { statusId: 1 }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+    const grantNumberLink = await GrantNumberLink.findOne({ where: { grantNumber: 'Grant100' } });
+    expect(grantNumberLink).not.toBe(null);
+
+    await MonitoringReviewGrantee.destroy({
+      where: { grantNumber: 'Grant100', granteeId: 'Grantee100', reviewId: 'Review100' },
+      fource: true,
+    });
+    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review100' }, fource: true });
+    await MonitoringGranteeLink.destroy({ where: { granteeId: 'Grantee100' }, fource: true });
+    await GrantNumberLink.destroy({ where: { grantNumber: 'Grant100' }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:455]:
```suggestion
      force: true,
    });
    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review100' }, force: true });
    await MonitoringGranteeLink.destroy({ where: { granteeId: 'Grantee100' }, force: true });
    await GrantNumberLink.destroy({ where: { grantNumber: 'Grant100' }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+
+    // Check associations
+    const statusLink = await MonitoringFindingHistoryStatusLink.findOne({ where: { statusId: 2 } });
+    expect(statusLink).not.toBe(null);
+
+    await MonitoringFindingHistoryStatus.destroy({
+      where: { statusId: 1 },
+      fource: true,
+    });
+    await MonitoringFindingHistoryStatusLink.destroy({ where: { statusId: 2 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:267]:
```suggestion
      force: true,
    });
    await MonitoringFindingHistoryStatusLink.destroy({ where: { statusId: 2 }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 9]
```diff
@@ -0,0 +1,542 @@
...
+    const findingLink = await MonitoringFindingLink.findOne({ where: { findingId: 'Finding001' } });
+    expect(findingLink).not.toBe(null);
+
+    await MonitoringFindingStandard.destroy({
+      where: { findingId: 'Finding001', standardId: 101 },
+      fource: true,
+    });
+    await MonitoringStandardLink.destroy({ where: { standardId: 101 }, fource: true });
+    await MonitoringStandardLink.destroy({ where: { standardId: 100 }, fource: true });
+    await MonitoringFindingLink.destroy({ where: { findingId: 'Finding001' }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:311]:
```suggestion
      force: true,
    });
    await MonitoringStandardLink.destroy({ where: { standardId: 101 }, force: true });
    await MonitoringStandardLink.destroy({ where: { standardId: 100 }, force: true });
    await MonitoringFindingLink.destroy({ where: { findingId: 'Finding001' }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 10]
```diff
@@ -0,0 +1,542 @@
...
+
+    // Check associations
+    const statusLink = await MonitoringFindingStatusLink.findOne({ where: { statusId: 3 } });
+    expect(statusLink).not.toBe(null);
+
+    await MonitoringFindingStatus.destroy({
+      where: { statusId: 3 },
+      fource: true,
+    });
+    await MonitoringFindingStatusLink.destroy({ where: { statusId: 3 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:349]:
```suggestion
      force: true,
    });
    await MonitoringFindingStatusLink.destroy({ where: { statusId: 3 }, force: true });
```

File: src/models/tests/monitoring/monitoring.test.js [hunk 11]
```diff
@@ -0,0 +1,542 @@
...
+
+    const statusLink = await MonitoringReviewStatusLink.findOne({ where: { statusId: 1 } });
+    expect(statusLink).not.toBe(null);
+
+    await MonitoringReview.destroy({
+      where: { reviewId: 'Review001', statusId: 1 },
+      fource: true,
+    });
+    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review001' }, fource: true });
+    await MonitoringReviewStatusLink.destroy({ where: { statusId: 1 }, fource: true });
```
Comment (hardwarehuman) [inline src/models/tests/monitoring/monitoring.test.js:403]:
```suggestion
      force: true,
    });
    await MonitoringReviewLink.destroy({ where: { reviewId: 'Review001' }, force: true });
    await MonitoringReviewStatusLink.destroy({ where: { statusId: 1 }, force: true });
```

File: src/models/hooks/genericLink.js [hunk 12]
```diff
@@ -41,8 +41,8 @@ const syncLink = async (
   // Generate a unique semaphore key based on the model name and entity ID
   const semaphoreKey = `${model.tableName}_${entityId}`;
   // Acquire a lock to ensure only one operation is performed on this entity at a time
-  await semaphore.acquire(semaphoreKey);
```
Comment (hardwarehuman) [inline src/models/hooks/genericLink.js:44]: Looks like a newline migrating from below to above the code that goes with the comment

**Review by hardwarehuman — APPROVED** [review]
Sorry about the code suggestions; I didn't realize how repetitive they would become. You can accept them all or do a global find/replace.

File: src/models/hooks/genericLink.js [hunk 13]
```diff
@@ -41,8 +41,8 @@ const syncLink = async (
...
```
Comment (GarrettEHill) [inline src/models/hooks/genericLink.js:44]:
```suggestion
```

File: src/models/hooks/genericLink.js [hunk 14]
```diff
@@ -41,8 +41,8 @@ const syncLink = async (
...
 
+  await semaphore.acquire(semaphoreKey);
```
Comment (GarrettEHill) [inline src/models/hooks/genericLink.js:45]:
```suggestion
  await semaphore.acquire(semaphoreKey);
```

**Pull request merged**